nemo_curator/modules/fuzzy_dedup.py

```diff
@@ -500,15 +510,26 @@ def __call__(self, dataset: DocumentDataset):
         if self.config.false_positive_check:
             # Map buckets to lower cardinality distribution
             print(f"Stage{stage_num} (False Positive Check): Starting Map_Buckets")
-            ddf_mapped_buckets_w_anchors = self.map_buckets.map_buckets_with_anchors(
-                documents_df=dataset.df, buckets_df=buckets_df.df
-            )
+            t0 = time.time()
             mapped_buckets_w_anchors_path = os.path.join(
                 self.config.cache_dir, "anchor_docs_with_bk.parquet"
             )
-            ddf_mapped_buckets_w_anchors.to_parquet(
-                mapped_buckets_w_anchors_path, write_index=False
+            with performance_report_if(
```

Comment:
For this profiling do we need _all_ of in this context or just the `to_parquet` part? Same questions below

Reply:
In general we need to wrap `performance_report` around any sections of code that trigger a computation. For most modules (such as `minhash/jaccard_compute` etc. everything is lazy until the `to_parquet` call, so wrapping just that call is sufficient to capture the whole profile. 

On the other hand, stages like `LSH` have a `persist` call in the code which trigger a computation. In this case wrapping just the `to_parquet` call will only wrap everything _after_ the `persist` since there has already been some computation before persist not captured by profiling.

Reply:
Good call out on intermediate _persists_, however if you see around Line 1466 where we call `from_dask_cudf_edgelist` and other methods within the `cc_workflow`. (or even in this method where we call `map_buckets_with_anchors`). Should we wrap these since it won't be obviously clear if within these methods there are intermediate actions happening (such as persist or writing to disk)